client.go

```diff
@@ -1014,9 +1058,9 @@ func (c *Client) makeHandledPromise() (*goja.Promise, func(interface{}), func(in
 		}
 }
 
-// connect establishes the client's connection to the target
+// Connect establishes the client's connection to the target
 // redis instance(s).
-func (c *Client) connect() error {
+func (c *Client) Connect() error {
```

Comment:
>the user must explicitly call the .connect method in the init context, or we keep connecting under the hood automatically?

We went with the second so do we need to export `connect`? Do we expect a use case for it? If yes, I think we should add a test for it and maybe expand a bit the documentation here.

Reply:
See me question in the PR description. The reason I export it is to be able to test it, as there are tests for it, at the end of the test file 😉 (that file is huge, I would have missed it myself).

The main rationale is: every method "connect" under the hood, to ensure that once in the VU context, the TLS config, and dialers are set up (it's forbidden to do that in the init context). To be able to test that connecting in the init context is forbidden, and that connecting to an unreachable server fails, I had to expose this method (so that I can test it on the JS side)

Reply:
>as there are tests for it, at the end of the test file wink (that file is huge, I would have missed it myself).
seen, thanks  :pray: 

>I had to expose this method (so that I can test it on the JS side)

It doesn't seem we really need the concrete JS runtime for those tests. The cases seem to me they are reproducible just using the `modulestest.NewRuntime(t)`. Am I wrong?

Reply:
You're right! That's super useful input, I'm going to integrate it, thanks a bunch 🙇🏻